### DISCUSSION TITLE
How does run_onepageos get called?

### DISCUSSION THREAD

**Discussion by nomadicinteractiveinc** [guysoft/FullPageOS#547] [Q&A]
Hi there,

Interesting project and one that can be of use to me.  Trying to figure out some of the anatomy. ;) 

I see `run_onepageos` runs, checks HTTP status and then runs the browser script if online.  The reload_fullpageos_txt is `killall /usr/lib/chromium-browser/chromium-browser`.  Considering this script is going to kill all instances of chromium, how is the run_onepageos script ran again?  Is there a service keeping it alive?

**Comment by guysoft** [comment]
FullPageOS uses the gui module of CustomPiOS.
The script is set as the start script here:
https://github.com/guysoft/FullPageOS/blob/devel/src/config#L11C1-L11C64

```bash
export GUI_STARTUP_SCRIPT=/opt/custompios/scripts/run_onepageos
```

This is used here:
https://github.com/guysoft/CustomPiOS/blob/devel/src/modules/gui/end_chroot_script#L17

```bash
# set the gui to run the main script of the GUI session
sed -i "s@GUI_SESSION_PLACEHOLDEFR@${GUI_STARTUP_SCRIPT}@g" /opt/custompios/scripts/start_gui
```

start_gui is set to be the session script in the gui:

https://github.com/guysoft/CustomPiOS/blob/devel/src/modules/gui/filesystem/root_init/usr/share/xsessions/guisession.desktop#L4

```
Exec=/opt/custompios/scripts/start_gui
```

The guisession.desktop is managed by XDG as far as i know
